lxd/storage/drivers/driver_zfs_utils.go

```diff
@@ -194,6 +194,40 @@ func (d *zfs) getDatasets(dataset string, types string) ([]string, error) {
 	return children, nil
 }
 
+func (d *zfs) filterRedundantOptions(dataset string, options ...string) ([]string, error) {
+	keys := []string{}
+	values := []string{}
+
+	// Extract keys and values from options.
+	for _, option := range options {
+		property := strings.Split(option, "=")
+
+		if len(property) != 2 {
+			continue
```

Comment:
Wouldn't it be safer to return an error here instead of silently skipping the wrongly formatted property? What do you think?

Reply:
Yes, I think you are right. I will change that, thanks!